### PR TITLE
Check juju and openstack command exist

### PR DESCRIPTION
**Pull request by xiywang**
If the commands are not exist, install before run the script.

Signed-off-by: Xiyue Wang <celia.wang@canonical.com>

### CODE REVIEW

File: juju-openstack-controller-example.sh [hunk 1]
```diff
@@ -7,6 +7,13 @@ fi
 
 set -ux
 
+type openstack
+if [ $? == 1 ]
```
Comment (thedac) [inline juju-openstack-controller-example.sh:11]: Our convention is to do the if, test, and then all on the same line:
if [sometest]; then

File: juju-openstack-controller-example.sh [hunk 2]
```diff
@@ -15,6 +22,12 @@ set -ux
 : ${MODEL_CONSTRAINTS:="virt-type=kvm"}
 : ${WORKSPACE:="/tmp"}
 
+type juju
+if [ $? == 1 ]
+then
```
Comment (thedac) [inline juju-openstack-controller-example.sh:27]: Same

**Review by thedac — CHANGES_REQUESTED** [review]
Celia,
Thanks for the PR!

I have a minor nit change request.

Since this script affects OSCI I am also going to ask for Ryan to review it for its functionality before landing.

**Review by ryan-beisner — CHANGES_REQUESTED** [review]
Thank you for the proposal.  However, I do not want to introduce commands which require sudo into this script.  That will fail in our CI as the CI user does not have the ability to sudo.  The dependencies are expected to be installed in advance (by whatever means the user chooses), and maybe we should say that clearly somewhere.

**Comment by lourot** [comment]
Closing for the reasons mentioned above, thanks!

**Pull request closed without merging**